src/models/tools/tool-button.ts

```diff
@@ -0,0 +1,40 @@
+import { getEnv, Instance, SnapshotOut, types } from "mobx-state-tree";
+import { getToolContentInfoByTool } from "./tool-content-info";
+
+const BaseToolButtonModel = types.model("BaseToolButton", {
```

Comment:
As long as you're moving the code around you might as well adopt the `.named().props()` convention for consistency with the others.

Reply:
Looking around more closely, it appears that the convention is that you/we use `.named().props()` for "derived" models, but not for "base" models, in which case you should leave it the way you have it.